### PR TITLE
TandemFoil Paper: lr=1.5e-4 at champion config

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Higher LR (1.5e-4 vs 1.25e-4) may converge faster. The TandemFoil parity benchmark found lr=1.25e-4 as the sweet spot, but TFP has different geometry (high-Re extrapolation tasks). Tests whether slightly higher LR helps TFP's longer convergence.

## Instructions

```bash
cd target/icml2026 && python train.py \
  --dataset tandemfoil_paper --optimizer lion --lr 1.5e-4 --cosine-t-max 10 \
  --grad-clip 0.5 --weight-decay 1e-2 --enable-fourier \
  --model-layers 3 --model-hidden-dim 192 --model-heads 3 \
  --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only \
  --asinh-pressure --residual-prediction --enable-pressure-prior-addition \
  --epochs 999 --ema-decay 0.999 \
  --wandb_group tfp-lr-sweep
```

## Reporting Requirements
- Best val `field_mse` + epoch
- Test `field_mse` at best val checkpoint
- W&B run ID

## Baseline

**TFP baselines:**
- val best: `field_mse = 0.002383` @ ep443 (PR #3025, W&B: `d1xh0o1p`)
- Config: Lion lr=1.25e-4, T_max=10, gc=0.5, EMA=0.999, 3L/192d, Fourier+physics
- paper target: normalized full-field MSE (field_mse, lower is better)
- **No clean test_primary/field_mse yet** — that is what the TFP experiments must establish

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT shouko:

## Results

### TFP lr=1.5e-4 — NEGATIVE RESULT (pressure divergence)

| Metric | Baseline (lr=1.25e-4, #3025) | This run (lr=1.5e-4) |
|---|---|---|
| val_primary/field_mse | **0.002383** @ ep443 | **Infinity** (all 71 epochs) |
| val/surface_mse_Ux | — | 0.0019 (ep60, healthy) |
| val/surface_mse_Uy | — | 0.0060 (ep60, healthy) |
| val/surface_mse_p | — | **Infinity** (71/71 epochs) |
| val/volume_mse_p | — | **Infinity** (71/71 epochs) |

**W&B run:** `tzwac8i0` (shouko/tfp-lr-15e4, group: tfp-lr-sweep)

**Command:**
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset tandemfoil_paper --optimizer lion --lr 1.5e-4 --cosine-t-max 10 \
  --grad-clip 0.5 --weight-decay 1e-2 --enable-fourier \
  --model-layers 3 --model-hidden-dim 192 --model-heads 3 \
  --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only \
  --asinh-pressure --residual-prediction --enable-pressure-prior-addition \
  --epochs 999 --ema-decay 0.999 \
  --agent shouko --wandb-name "shouko/tfp-lr-15e4" --wandb-group tfp-lr-sweep
```

### What happened

**Hypothesis falsified.** lr=1.5e-4 causes complete pressure channel divergence on TFP. The velocity components (Ux, Uy) converge normally — surface_mse_Ux drops from ~0.008 to 0.0019, surface_mse_Uy from ~0.019 to 0.006 — but the pressure channel (surface_mse_p) was Infinity for every single epoch. The pressure briefly showed finite (but astronomical: ~1e+20 to 1e+32) values around ep11–40, then diverged back to true Infinity by ep50.

Training loss decreased steadily (0.98→0.057 at ep60) then showed instability spikes at ep65-68 (up to 0.111), suggesting the optimizer is starting to lose stability overall.

The pressure channel in TFP is extremely sensitive to learning rate. The jump from lr=1.25e-4 (which produced field_mse=0.002383) to lr=1.5e-4 (which produced permanent Infinity) is only a 20% LR increase, yet it completely breaks pressure convergence. This is likely because the `asinh-pressure` inverse transform amplifies pressure residuals, and the slightly higher LR pushes gradients past the point where gc=0.5 can stabilize them.

**Key insight:** lr=1.25e-4 is very close to the upper stability boundary for TFP pressure with this config. Any LR sweep for TFP should go downward (lr=1e-4, 7.5e-5) rather than upward.

### Suggested follow-ups
- lr=1e-4 or lr=7.5e-5 sweep downward to find TFP's LR sweet spot
- gc=0.3 at lr=1.25e-4 to test if tighter clipping can handle higher LR
- Separate pressure LR (lower) from velocity LR if the architecture supports per-channel LR scaling

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #3097 because ADVISOR: Closing — lr=1.5e-4 is above the TFP stability boundary.

**Results:** val_primary/field_mse = Infinity for all 89 epochs. Pressure channel never produced finite values. Velocity channels (Ux, Uy) were healthy, confirming this is a pressure-specific LR instability.

**Key signal:** 100% gradient clip rate across all 89 epochs — the optimizer was perpetually saturating the gc=0.5 clip threshold. lr=1.25e-4 is confirmed at or near the Lion+gc=0.5 stability ceiling for TFP pressure.

**Conclusion:** LR above 1.25e-4 is a dead end for TFP with current config. shinobu (#3096) testing lr=1e-4 (downward) is the right direction. Your suggestion of separate pressure/velocity LR is interesting but not ICML-sprint critical.

Reassigning to DrivAerML (top priority)..
